services/ui-src/src/components/reports/ModalDrawerReportPage.tsx

```diff
@@ -1,107 +1,61 @@
+  const {
+    isOpen: addEditEntityModalIsOpen,
+    onOpen: addEditEntityModalOnOpenHandler,
+    onClose: addEditEntityModalOnCloseHandler,
+  } = useDisclosure();
 
-  const openRowDrawer = (entity: string) => {
-    setCurrentEntity(entity);
-    onOpen();
-  };
-
-  const removeEntity = (entityTitle: string) => {
-    setEntities(entities.filter((entity) => entity.title !== entityTitle));
-  };
-
-  const onSubmit = async (formData: AnyObject) => {
-    if (userIsStateUser || userIsStateRep) {
-      setSubmitting(true);
-      const reportKeys = {
-        state: state,
-        id: report?.id,
-      };
-      const dataToWrite = {
-        status: ReportStatus.IN_PROGRESS,
-        lastAlteredBy: full_name,
-        fieldData: formData,
-      };
-      await updateReport(reportKeys, dataToWrite);
-      setSubmitting(false);
+  const openAddEditEntityModal = (entityId?: string) => {
+    if (report && entityId) {
+      // pre-fill form if if editing an existing entity
+      setSelectedEntity(report.fieldData[entityType]);
     }
+    // use disclosure to open modal
+    addEditEntityModalOnOpenHandler();
   };
 
   return (
     <Box data-testid="dynamic-drawer-section">
-      {entities.length > 0 && (
-        <Box style={sx.entityTable}>
-          <Heading as="h3" sx={sx.tableTitle}>
-            {route.drawer.title}
-          </Heading>
-          {entities.map((entity) => {
-            return (
-              <Flex key={entity.title} sx={sx.entityRow}>
-                <Heading as="h4" sx={sx.entityName}>
-                  {entity.title}
-                </Heading>
-                <Flex key={entity.title} sx={sx.buttonContainer}>
-                  <Button
-                    sx={sx.buttonStyle}
-                    onClick={() => openRowDrawer(entity.title)}
-                    variant="outline"
-                  >
-                    Edit
-                  </Button>
-                  <Button
-                    sx={sx.buttonStyle}
-                    onClick={() => removeEntity(entity.title)}
-                    variant="outline"
-                  >
-                    Delete
-                  </Button>
-                </Flex>
-              </Flex>
-            );
-          })}
-        </Box>
-      )}
-      <Button onClick={() => openRowDrawer("")} type="submit">
-        Add TEMPORARY
-      </Button>
-      <ReportDrawer
-        drawerDisclosure={{
-          isOpen,
-          onClose,
-        }}
-        drawerTitle={currentEntity}
-        form={route.drawer.form}
-        onSubmit={onSubmit}
-        data-testid="report-drawer"
-      />
+      <Box>
+        <Button
+          sx={sx.addEntityButton}
+          onClick={() => openAddEditEntityModal()}
+        >
+          {dashboard.addEntityButtonText}
+        </Button>
+        {entities && (
+          <>
+            <Heading as="h4" sx={sx.dashboardTitle}>
+              {dashboard.title}
+            </Heading>
```

Comment:
So entities being an empty array will return this title, just double checking that that is intentional and if it is, do we need this check first since the heading will always show?

Reply:
ooh yeah good catch. i hadn't considered that with delete in place there's a possibility it could be `[]`. changed it so on line 12 it returns either the result or an empty array, and you're right: now we don't need the check. 👍